core/src/kiso.rs

```diff
@@ -0,0 +1,143 @@
+//! Actor responsible for configuration state and its dynamic updates.
+//!
+//! Currently the API exposed by [`KisoHandle`] works only with [`ConfigurationDTO`], because
+//! no any part of Iroha is interested in the whole state. However, the API could be extended
+//! in future.
+//!
+//! ## Example
+//!
+//! ```
+//! #[tokio::main]
+//! async fn main() {
+//!   todo!()
+//! }
+//! ```
+
+use eyre::Result;
+use iroha_config::{
+    client_api::{ConfigurationDTO, Logger as LoggerDTO},
+    iroha::Configuration,
+};
+use iroha_logger::actor::{Error as LoggerError, LoggerHandle};
+use tokio::sync::{mpsc, oneshot};
+
+const DEFAULT_CHANNEL_SIZE: usize = 32;
+
+/// The handle to work with the actor.
+///
+/// The actor will shutdown when all its handles are dropped.
+#[derive(Clone)]
+pub struct KisoHandle {
+    sender: mpsc::Sender<Message>,
+}
+
+impl KisoHandle {
+    /// Spawn a new actor
+    pub fn new(state: Configuration, logger: LoggerHandle) -> Self {
+        let (sender, receiver) = mpsc::channel(DEFAULT_CHANNEL_SIZE);
+        let mut actor = Actor {
+            receiver,
+            state,
+            logger,
+        };
+        tokio::spawn(async move { actor.run().await });
+
+        Self { sender }
+    }
+
+    /// Fetch the [`ConfigurationDTO`] from the actor's state.
+    ///
+    /// # Errors
+    /// If communication with actor fails.
+    pub async fn get_dto(&self) -> Result<ConfigurationDTO, Error> {
```

Comment:
Why do you choose message sending to propagate configuration? 

Some time ago i discovered [`arc_swap` ](https://github.com/vorner/arc-swap) create, from it's [usage patterns](https://docs.rs/arc-swap/latest/arc_swap/docs/patterns/index.html) i can see some benefits:

- it would allow to decouple configuration actor and configuration, so every configuration user will just have shared configuration type (or wrapper around it), not actor handle
- reduced amount of message sending to load configuration

Reply:
`arc_swap` seems to be a good option in case of sharing mutable state.

Some notes:

- Actually, now the config itself is almost not shared. The initial values are read and directly stored in actors, and only `Torii` holds the whole structure and interacts with it.
- We have a stateful system with many actors, which is not the same as patterns described in `arc_swap`. If in future we add more dynamic configurations related to internal logic of actors, it wouldn't be enough to just share the state: we will need a notification mechanism so that actors might react to changes. `Kiso` might propagate updates naturally.
- With `arc_swap`, each holder of the config will be able to not only read, but also write changes. It makes mutations decentralised and the architecture might become confusing to maintain and change, some subtle semantic bugs might appear etc. A central responsible point makes things clear, and it seems for me not achievable with `arc_swap`'s design.
- Shared mutable state is not idiomatic in the actor model.
- If in future we will need to share big parts of the config frequently, it is still achievable with `Kiso` and careful design of caching and notifications.
- Currently all dynamic configuration interactions happen only during configuration endpoints handling, which I guess is not that part that will be invoked frequently. So, I don't think message passing overhead is significant. Anyway, caching might always be used on top of that in future.

Reply:
Thanks for detailed explanation, your arguments seem reasonable to me